File: BinaryTreePaths/BinaryTreePaths.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def binaryTreePaths(self, root):
        """
        :type root: TreeNode
        :rtype: List[str]
        """
        if root is None:
            return []
        res = []
        dfs_Path(root, '', res)
        return res

def dfs_Path(root, tmp, res):
    if root.left:
        dfs_Path(root.left, tmp + str(root.val) + '->', res)
    if root.right:
        dfs_Path(root.right, tmp + str(root.val) + '->', res)
    if root.left is None and root.right is None:
        tmp += str(root.val)
        res += tmp,
        return

File: BinaryTreePaths/BinaryTreePaths.cpp
#include <vector>
#include <string>

using namespace std;
/**
* Definition for a binary tree node.
*/
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

class Solution {
public:
	vector<string> binaryTreePaths(TreeNode* root) {
		vector<string> result;
		if (root == nullptr)
			return result;
		string tmp = to_string(root->val);
		DFS(root, result, tmp);
		return result;
	}

private:
	void DFS(TreeNode* node, vector<string> &result, string path) {
		if (node->left) {
			DFS(node->left, result, path + "->" + to_string(node->left->val));
		}
		if (node->right) {
			DFS(node->right, result, path + "->" + to_string(node->right->val));
		}
		if (!node->left && !node->right) {
			result.push_back(path);
			return;
		}
	}
};